common-libs/common/src/services/DidDocumentService/LocalKeyVault.ts

```diff
@@ -0,0 +1,46 @@
+import { KeyVault } from './KeyVault'
+import { KeysService } from '../index'
+import * as secp256k1 from 'secp256k1'
+
+/**
+ * KeyStore backed implementation when the seed is known
+ */
+export class LocalKeyVault implements KeyVault {
+  private _keyService: KeysService
+
+  constructor(keyService: KeysService) {
+    this._keyService = keyService
+  }
+
+  primaryPublicKey(): Buffer {
+    const { seed } = this._keyService.decryptSeed()
+    const seedHex = seed.toString('hex')
+
+    const { publicKey } = KeysService.getPublicAndPrivateKeys(seedHex, 'elem')
+    return publicKey
+  }
+
+  recoveryPublicKey(): Buffer {
+    const { seed } = this._keyService.decryptSeed()
+    const seedHex = seed.toString('hex')
+
+    const { publicKey } = KeysService.getAnchorTransactionPublicAndPrivateKeys(seedHex, 'elem')
+    return publicKey
+  }
+
+  externalKeys(): any[] {
+    const { externalKeys } = this._keyService.decryptSeed()
+    return externalKeys
+  }
+
+  sign(payload: Buffer): Buffer {
+    const { seed } = this._keyService.decryptSeed()
+    const seedHex = seed.toString('hex')
+
+    const { privateKey } = KeysService.getPublicAndPrivateKeys(seedHex, 'elem')
```

Comment:
We might want to cache the result of `getPublicAndPrivateKey` in this class. This operation seems to be somewhat time-consuming; and these changes introduce performance degradation, as we only called `getPublicAndPrivateKeys` once in the original version.

Reply:
I'm closing this as it's similar to https://github.com/affinityproject/affinidi-core-sdk/pull/129#discussion_r692002993